src/creatwth.cpp

```diff
@@ -499,37 +508,57 @@ static BOOL UpdateFrom32To64(HANDLE hProcess, HMODULE hModule, WORD machine,
 }
 #endif // DETOURS_64BIT
 
-typedef BOOL(WINAPI *LPFN_ISWOW64PROCESS)(HANDLE, PBOOL);
+namespace Detour {
 
-static BOOL IsWow64ProcessHelper(HANDLE hProcess,
-                                 PBOOL Wow64Process)
-{
-#ifdef _X86_
-    if (Wow64Process == NULL) {
-        return FALSE;
-    }
+#ifndef PROCESSOR_ARCHITECTURE_AMD64
+#define PROCESSOR_ARCHITECTURE_AMD64            9
+#endif
 
-    // IsWow64Process is not available on all supported versions of Windows.
-    //
-    HMODULE hKernel32 = LoadLibraryW(L"KERNEL32.DLL");
-    if (hKernel32 == NULL) {
-        DETOUR_TRACE(("LoadLibraryW failed: %d\n", GetLastError()));
-        return FALSE;
-    }
+#ifndef PROCESSOR_ARCHITECTURE_ARM64
+#define PROCESSOR_ARCHITECTURE_ARM64            12
+#endif
 
-    LPFN_ISWOW64PROCESS pfnIsWow64Process = (LPFN_ISWOW64PROCESS)GetProcAddress(
-        hKernel32, "IsWow64Process");
+    BOOL Is64BitOS()
+    {
+        BOOL bRet = FALSE;
+        HMODULE hModule = GetModuleHandle(TEXT("kernel32.dll"));
+        if (!hModule) {
+            return bRet;
+        }
+        VOID(WINAPI * _GetNativeSystemInfo)(OUT LPSYSTEM_INFO lpSystemInfo) = (void(__stdcall*)(LPSYSTEM_INFO))GetProcAddress(hModule, "GetNativeSystemInfo");
+        if (!_GetNativeSystemInfo) {
+            return bRet;
+        }
 
-    if (pfnIsWow64Process == NULL) {
-        DETOUR_TRACE(("GetProcAddress failed: %d\n", GetLastError()));
-        return FALSE;
+        SYSTEM_INFO si;
+        _GetNativeSystemInfo(&si);
+        if (si.wProcessorArchitecture == PROCESSOR_ARCHITECTURE_AMD64 || si.wProcessorArchitecture == PROCESSOR_ARCHITECTURE_ARM64 ||
+            si.wProcessorArchitecture == PROCESSOR_ARCHITECTURE_IA64 || si.wProcessorArchitecture == PROCESSOR_ARCHITECTURE_ALPHA64) {
```

Comment:
>| si.wProcessorArchitecture == PROCESSOR_ARCHITECTURE_ALPHA64 [](start = 70, length = 61)

Detours doesn't support ALPHA, no point in checking

Reply:
Is64BitOS exists as a general-purpose function. It is ported from my private library. Detours users can also copy and reuse it in other codes. It is judged that all current 64-bit Windows operating systems conform to the meaning of the function name, and here, although Detours does not support the ALPHA/ALPHA64 architecture, the judgment of this condition does not affect it. Because it just tells the caller whether the current system is 64-bit. Whether the current process is 32-bit is also determined by the Is64BitProcess function.